### DISCUSSION TITLE
Need help with Bitbucket pipelines

### DISCUSSION THREAD

**Discussion by levipadre** [vercel/community#2691] [Help]
### Problem Description

I try to write a Bitbucket pipeline, which does the following when I push the code to master.

- deploy my Nextjs project (App Router) on Vercel, but only create a preview site
- run tests on this preview site, e.g. I have some scripts in my package.json
- deploy to master/production if all tests passed

I've set up `VERCEL_TOKEN`, `VERCEL_ORG_ID` and `VERCEL_PROJECT_ID` keys already in Bitbucket, but I can't get my head around the pipeline.

**Comment by amyegan** [comment]
Hi @levipadre! We have a guide with an example Bitbucket Pipeline script. That's a good place to start with Vercel deployments via Pipelines. You can find the guide here: https://vercel.com/guides/how-can-i-use-bitbucket-pipelines-with-vercel

For other elements of your workflow, like running tests, you may need to get some guidance from the Bitbucket folks or advice from people who build the testing tool used in your project.
